### PR TITLE
add temporal difference encoder module and tutorial

### PR DESCRIPTION
**Pull request by AI-pha**
This is the spiking temporal difference encoder that we wrote in Telluride last year.

### CODE REVIEW

File: nxsdk_modules_contrib/temporal_diff_enc/stde.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
+# Copyright(c) 2020 University of Zurich. All rights reserved.
```
Comment (srrisbud) [inline nxsdk_modules_contrib/temporal_diff_enc/stde.py:1]: Please remove the copyright and license information from all files. We include a LICENSE https://github.com/intel-nrc-ecosystem/models/blob/master/LICENSE.

Instead please modify the README https://github.com/intel-nrc-ecosystem/models/blob/master/README.md Acknowledgement Section with your network module and authorship. You may add any other third party acknowledgements and citations in a local README in your module.

**Comment by AI-pha** [comment]
Thanks, I can do that. Maybe you could specify this more clearly in the CONTRIBUTING.md.
There you say: "All code submitted to the project is required to carry that license."
And the other code in the repo actually contains the license in the header, so it is a bit unclear.
I guess "carry the license" is meant in a figurative sense?

**Comment by AI-pha** [comment]
Hi, could you have a look at my pull request again?
There seems to be 1 pending reviewer and a slightly unclear situation concerning the inclusion of the license.

**Review by joyeshmishra — APPROVED** [review]

**Comment by joyeshmishra** [comment]
> Hi, could you have a look at my pull request again?
> There seems to be 1 pending reviewer and a slightly unclear situation concerning the inclusion of the license.

Alpha -- Sorry, could you please explain what's unclear to Sumedh's response?

**Comment by AI-pha** [comment]
> Thanks, I can do that. Maybe you could specify this more clearly in the CONTRIBUTING.md.
> There you say: "All code submitted to the project is required to carry that license."
> And the other code in the repo actually contains the license in the header, so it is a bit unclear.
> I guess "carry the license" is meant in a figurative sense?

Thanks, the response itself is quite clear, and I understand what needs to be done. What makes it unclear is that it is seemingly in contradiction with the CONTRIBUTING.md (as detailed in my earlier comment).
Some additional context:
I just tried to replicate the current practice in the files contributed by Intel authors, but there are obviously different guidelines for non-Intel authors, and I think they should be specified in the CONTRIBUTING.md.

**Pull request merged**